src/PrestaShopBundle/Entity/Repository/LangRepository.php

```diff
@@ -26,24 +26,85 @@
 
 namespace PrestaShopBundle\Entity\Repository;
 
+use Doctrine\ORM\EntityManager;
 use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\Mapping;
+use PrestaShop\PrestaShop\Core\Language\LanguageRepositoryInterface;
+use PrestaShopBundle\Entity\Lang;
 
-class LangRepository extends EntityRepository
+class LangRepository extends EntityRepository implements LanguageRepositoryInterface
 {
+    const ISO_CODE = 'isoCode';
+    const LOCALE = 'locale';
+
+    /**
+     * Stores language instances in different arrays to match them quickly
+     * via a criteria and avoid multiple database queries.
+     *
+     * @var array
+     */
+    private $matches;
+
+    /**
+     * @param EntityManager $em
+     * @param Mapping\ClassMetadata $class
+     */
+    public function __construct(EntityManager $em, Mapping\ClassMetadata $class)
```

Comment:
Useless constructor if you set values in properties. and don't use static.

Reply:
right!